great_expectations/data_context/data_context/context_factory.py

```diff
@@ -37,11 +37,300 @@
         EphemeralDataContext,
         FileDataContext,
     )
+    from great_expectations.data_context.store import Store
     from great_expectations.data_context.types.base import DataContextConfig
 
 ContextModes: TypeAlias = Literal["file", "cloud", "ephemeral"]
 
 
+class ProjectManager:
+    """Singleton class to manage projects in the global namespace."""
+
+    _project: AbstractDataContext | None
+
+    def __init__(self):
+        self._project = None
+
+    def get_project(  # noqa: PLR0913
+        self,
+        project_config: DataContextConfig | Mapping | None = None,
+        context_root_dir: PathStr | None = None,
+        project_root_dir: PathStr | None = None,
+        runtime_environment: dict | None = None,
+        cloud_base_url: str | None = None,
+        cloud_access_token: str | None = None,
+        cloud_organization_id: str | None = None,
+        cloud_mode: bool | None = None,
+        mode: ContextModes | None = None,
+    ) -> AbstractDataContext:
+        self._project = self._build_context(
+            project_config=project_config,
+            context_root_dir=context_root_dir,
+            project_root_dir=project_root_dir,
+            runtime_environment=runtime_environment,
+            cloud_base_url=cloud_base_url,
+            cloud_access_token=cloud_access_token,
+            cloud_organization_id=cloud_organization_id,
+            cloud_mode=cloud_mode,
+            mode=mode,
+        )
+        return self._project
+
+    def set_project(self, project: AbstractDataContext) -> None:
+        self._project = project
+
+    def get_store(self, model: Any) -> Store:
+        if not self._project:
+            raise RuntimeError(
+                "DataContext must be initialized before using `get_store`."
+            )
+        # todo: do something more elegant here
+        from great_expectations.core import ExpectationSuite
+
+        if isinstance(model, ExpectationSuite):
+            return self._project.expectations_store
+        else:
+            raise NotImplementedError
```

Comment:
Could we just have a map between type and store? Could be a class level attr

Reply:
Agreed that it's fine to defer for now, but I'd definitely rather we just expose top-level `<MODEL>_store` properties / methods.

Reply:
@tyler-hoffman so models would import the `project_manager` instance and access the property there? or get a copy of the context and access the property from there?